Source/CarthageKit/Project.swift

```diff
@@ -559,6 +598,25 @@ private func frameworksInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL,
 		}
 }
 
+/// Sends the URL to each dSYM found in the given directory
+private func dSYMsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL, "com.apple.xcode.dsym")
+}
+
+/// Sends the URL of the dSYM whose UUIDs match those of the given framework, or
+/// errors if there was an error parsing a dSYM contained within the directory.
+private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return combineLatest(UUIDsForFramework(frameworkURL), dSYMsInDirectory(directoryURL))
```

Comment:
Good catch! I only tested it on an archive that had one dSYM, not two. The issue and fix both make sense—I'll update the diff.